client/jetpack-connect/index.js

```diff
@@ -17,16 +17,12 @@ const redirectToStoreWithInterval = context => {
 };
 
 export default function() {
-	page( '/jetpack/connect/install', controller.install );
+	page( '/jetpack/connect/:type(personal|premium|pro)/:interval(yearly|monthly)?', controller.connect );
```

Comment:
I much prefer the name `interval` that you have used here, but I note that there are quite a few places where the name `intervalType` has been used instead. I think I favour using `interval`, and maybe cleaning up the uses of `intervalType` to match later on.